data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -0,0 +1,236 @@
+        }
+        return inputSerialization;
+    }
+
+    private InputStream getInputStreamFromResponseHeader(final S3SelectResponseHandler responseHand) {
+        return responseHand.getReceivedEvents().stream()
+                .filter(e -> e.sdkEventType() == SelectObjectContentEventStream.EventType.RECORDS)
+                .map(e -> ((RecordsEvent) e).payload().asInputStream()).collect(Collectors.toList()).stream()
+                .reduce(SequenceInputStream::new).orElse(null);
+    }
+
+    private SelectObjectContentRequest getS3SelectObjRequest(final S3ObjectReference s3ObjectReference,
+            final InputSerialization inputSerialization) {
+        return SelectObjectContentRequest.builder().bucket(s3ObjectReference.getBucketName())
+                .key(s3ObjectReference.getKey()).expressionType(ExpressionType.SQL)
+                .expression(queryStatement)
+                .inputSerialization(inputSerialization)
+                .outputSerialization(outputSerialization -> outputSerialization.json(SdkBuilder::build))
+                .build();
+    }
+
+    private long parseCompleteStreamFromResponseHeader(final S3ObjectReference s3ObjectReference,
+            final BufferAccumulator<Record<Event>> bufferAccumulator,final InputStream stream) throws IOException {
+        final AtomicLong totalBytesRead = new AtomicLong(0);
+        try (final CountingInputStream countingInputStream = new CountingInputStream(stream);
+             final InputStreamReader inputStreamReader = new InputStreamReader(countingInputStream);
+             final BufferedReader reader = new BufferedReader(inputStreamReader)) {
+            Optional<String> selectObjectOptionalString;
+            while ((selectObjectOptionalString = Optional.ofNullable(reader.readLine())).isPresent()) {
+                final Optional<JsonNode> optionalNode = addingS3Reference(s3ObjectReference,
+                        getJsonNode(selectObjectOptionalString.get()));
+                if (optionalNode.isPresent()){
+                    Record<Event> eventRecord = new Record<>(JacksonLog.builder().withData(optionalNode.get()).build());
+                    try{
+                        eventConsumer.accept(eventRecord.getData(),s3ObjectReference);
+                        bufferAccumulator.add(eventRecord);
+                    }catch(final Exception ex){
+                        throw new IOException(ex);
+                    }
+
+                }
+            }
+            totalBytesRead.addAndGet(countingInputStream.getByteCount());
+        }
+        try {
+            bufferAccumulator.flush();
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+        return totalBytesRead.get();
+    }
+
+    private JsonNode getJsonNode(String selectObjectOptionalString) throws JsonProcessingException {
+        final JsonNode recordJsonNode;
+        if(serializationFormatOption.equals(S3SelectSerializationFormatOption.CSV) &&
```

Comment:
I'm not sure I understand this conditional. It appears that we always use JSON as the output format. So why have this condition? How does the CSV come into play here?

Reply:
Thanks for your Review, When we have csv file header type to NONE,  S3 select service returning JSON with the below mentioned format. To make this consistent output I have added the csv processing code which will replace the s3 select output(_1,_2,_3,_4) key to the column name(column1, column2, column3, column4).

**S3 Select Output for CSV:**
{
  "_1": "2022-06-30",
  "_2": "15:03:52",
  "_3": "DEL26",
  "_4": "4628"
}

**Existing codec Output for CSV:**
{
"column1":"2022-06-30",
"column2":"15:03:52",
"column3":"DEL26",
"column4":"4628"
}